controller/array_action/ds8k_rest_client.py

```diff
@@ -0,0 +1,297 @@
+    # There are two style lun number, one's decimal value is <256 and the other
+    # is full as 16 hex digit. According to T10 SAN, when decimal value value
+    # is more then 256 and it is converted to the full 16 hex digit, it should
+    # be swapped and hex.
+    # For example, zlinux lun number stored in SC DB is 1075331113. It should
+    # be hex to '40184029' and swapped to '40294018'. When the lun number is 12
+    # and it is <256, it should be hex to '0c' directly.
+    pretreated_lun = int(lun)
+    if pretreated_lun < 256:
+        return _int_lunid_to_hex(pretreated_lun)
+    else:
+        return '{0:x}'.format(
+            (pretreated_lun >> 16 & 0xFFFF) | (pretreated_lun & 0xFFFF) << 16
+        )
+
+
+def scsilun_to_int(lun):
+    # There are two style lun number, one's decimal value is <256 and the other
+    # is full as 16 hex digit. According to T10 SAN, the full 16 hex digit
+    # should be swapped and converted into decimal.
+    # For example, SC got zlinux lun number from DS8K API, '40294018'. And it
+    # should be swapped to '40184029' and converted into decimal, 1075331113.
+    # When the lun number is '0c' and its decimal value is <256, it should be
+    # converted directly into decimal, 12.
+    pretreated_scsilun = int(lun, 16)
+    if pretreated_scsilun < 256:
+        return pretreated_scsilun
+    else:
+        return (pretreated_scsilun >> 16 & 0xFFFF) | \
+               (pretreated_scsilun & 0xFFFF) << 16
+
+
+class RESTClient(object):
+    """
+    driver side client. Used to interaction with pyds8k client as an adaptor.
+
+    !--important: the id field of all resources are case insensitive--!
+    """
+
+    def __init__(self, service_address, user, password,
+                 port=None,
+                 hostname='',
+                 ):
+        client_kwargs = {'service_address': service_address,
+                         'user': user,
+                         'password': password,
+                         }
+        if port:
+            client_kwargs.update({'port': port})
+        if hostname:
+            client_kwargs.update({'hostname': hostname})
+
+        self.client = Client(**client_kwargs)
+
+    @dictionarize
+    def get_system(self):
+        return self.client.get_system()
+
+    @dictionarize
+    def get_volume(self, volume_id):
+        return self.client.get_volumes(volume_id)
+
+    @dictionarize
+    def get_host(self, host_name):
+        return self.client.get_hosts(host_name)
+
+    @dictionarize
+    def get_host_mapping(self, host_name, lunid):
+        return self.client.get_mapping_by_host(host_name=host_name,
+                                               lunid=int_to_scsilun(lunid)
+                                               )
+
+    @dictionarize
+    def get_host_port(self, port_id):
+        return self.client.get_host_ports(port_id)
+
+    @dictionarize
+    def get_extentpool(self, pool_id):
+        pool = self.client.get_pools(pool_id)
+        try:
```

Comment:
exception are usually used to handle errors, not to handle logic.

Reply:
I think it can also be used to handle logic and it is more efficient, at least it doesn't need to check if it has attr "representation" every time. Usually it has this attr.